app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeavePresenter.java

```diff
@@ -72,98 +70,128 @@ public void onRequestError(BaseException error) {
-                validateCompensationFrom(request.getCompensation().getFromTime());
-                validateCompensationTo(request.getCompensation().getToTime());
-                validateReason(request.getReason());
+            case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_A:
+            case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_M:
+                if (validateCheckoutTime(request.getCheckoutTime())
+                        & validateCompensationFrom(request.getCompensation().getFromTime())
+                        & validateCompensationTo(request.getCompensation().getToTime())
+                        & validateReason(request.getReason())
+                        & validateProjectName(request.getProject())) {
+                    return true;
+                }
                 break;
-            case Constant.LeaveType.IN_LATE_WOMAN_A:
-            case Constant.LeaveType.IN_LATE_WOMAN_M:
-                validateCheckinTime(request.getCheckinTime());
+            case RequestLeaveViewModel.LeaveType.IN_LATE_WOMAN_A:
+            case RequestLeaveViewModel.LeaveType.IN_LATE_WOMAN_M:
+                if (validateCheckinTime(request.getCheckinTime()) & validateProjectName(
+                        request.getProject())) {
+                    return true;
+                }
                 break;
-            case Constant.LeaveType.LEAVE_EARLY_WOMAN_A:
-            case Constant.LeaveType.LEAVE_EARLY_WOMAN_M:
-                validateCheckoutTime(request.getCheckoutTime());
+            case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_WOMAN_A:
+            case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_WOMAN_M:
+                if (validateCheckoutTime(request.getCheckoutTime()) & validateProjectName(
+                        request.getProject())) {
+                    return true;
+                }
                 break;
-            case Constant.LeaveType.FORGOT_CARD_ALL_DAY:
-            case Constant.LeaveType.FORGOT_CHECK_ALL_DAY:
-                validateCheckinTime(request.getCheckinTime());
-                validateCheckoutTime(request.getCheckoutTime());
+            case RequestLeaveViewModel.LeaveType.FORGOT_CARD_ALL_DAY:
+            case RequestLeaveViewModel.LeaveType.FORGOT_CHECK_ALL_DAY:
+                if (validateCheckinTime(request.getCheckinTime()) & validateCheckoutTime(
+                        request.getCheckoutTime()) & validateProjectName(request.getProject())) {
+                    return true;
+                }
+
                 break;
-            case Constant.LeaveType.FORGOT_CARD_IN:
-            case Constant.LeaveType.FORGOT_TO_CHECK_IN:
-                validateCheckinTime(request.getCheckinTime());
+            case RequestLeaveViewModel.LeaveType.FORGOT_CARD_IN:
+            case RequestLeaveViewModel.LeaveType.FORGOT_TO_CHECK_IN:
+                if (validateCheckinTime(request.getCheckinTime()) & validateProjectName(
```

Comment:
ERROR: [STYLE][NS_DANGEROUS_NON_SHORT_CIRCUIT] Potentially dangerous use of non-short-circuit logic in com.framgia.wsm.screen.requestleave.RequestLeavePresenter.validateDataInput(Request)
http://findbugs.sourceforge.net/bugDescriptions.html#NS_DANGEROUS_NON_SHORT_CIRCUIT